solr/solrj/src/java/org/apache/solr/client/solrj/beans/DocumentObjectBinder.java

```diff
@@ -408,28 +406,28 @@ private Object getFieldValue(SolrDocument solrDocument) {
       }
     }
 
-    @SuppressWarnings({"unchecked", "rawtypes"})
     <T> void inject(T obj, SolrDocument sdoc) {
       Object val = getFieldValue(sdoc);
       if(val == null) {
         return;
       }
 
       if (isArray && !isContainedInMap) {
-        List list;
+        List<?> list;
         if (val.getClass().isArray()) {
           set(obj, val);
           return;
         } else if (val instanceof List) {
-          list = (List) val;
+          list = (List<?>) val;
         } else {
-          list = new ArrayList();
-          list.add(val);
+          List<Object> temp = new ArrayList<>();
+          temp.add(val);
+          list = temp;
```

Comment:
In this case I think we can actually get away with Collections.singletonList because we turn it into an array on the next line anyway. In general, it's because you can't add any objects to a wildcard capture.